spec/ics-027-interchain-accounts/README.md

```diff
@@ -210,54 +233,94 @@ function onChanCloseConfirm(
 
 In plain English, between chains `A` and `B`. It will describe only the case that chain A wants to register an Interchain account on chain B and control it. Moreover, this system can also be applied the other way around.
 
+`onRecvPacket` is called by the routing module when a packet addressed to this module has been received.
+
 ```typescript
-function onRecvPacket(packet: Packet): bytes {
-  if (packet.data is RunTxPacketData) {
+function onRecvPacket(packet: Packet) {
+  IBCAccountPacketData data = packet.data
+
+  switch (data.type) {
+  case Type.REGISTER:
+    let address
+    try {
+      // Create an account by using the packet's data (destination port, destination channel, etc) and packet data's data.
+      address = createAccount(packet, data.data)
+    } catch (e) {
+      // Return ack with error.
+      return IBCAccountPacketAcknowledgement{
+        type: Type.REGISTER,
+        chainID,
+        code: 1,
+        data: [],
+        error: e.message,
+      }
+    }
+    // Return ack with generated address.
+    return IBCAccountPacketAcknowledgement{
```

Comment:
Put this in the `try` block, it's cleaner flow-wise

Reply:
Ok

Reply:
I fixed it.